src/main/java/training/endava/playground/generics/types/LimitedContainer.java

```diff
@@ -0,0 +1,27 @@
+package training.endava.playground.generics.types;
+
+import java.util.ArrayList;
+
+public class LimitedContainer<T, N extends Number> {
+
+    private ArrayList<T> contents;
+
+    private N limit;
+
+    public LimitedContainer(N limit)
+    {
+        contents = new ArrayList<>();
+        this.limit = limit;
+    }
+
+    public void addContainer(T item)
+    {
+        if(this.contents.size() < (Integer) limit){
```

Comment:
this is not the generic way of comparing two generic types, if for example the generic argument would be a Double or anything else you would get a runtime class cast exception
try smth like that
https://stackoverflow.com/questions/2683202/comparing-the-values-of-two-generic-numbers